src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,182 @@
+            }
+        }
+        printDividerLine();
+    }
+
+    public void addTask(String line) {
+        String[] keywords = line.split(" ", 2);
+        boolean hasNoException = false;
+        boolean isEmptyDescription = true;
+
+        if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("todo")) {
+            try {
+                tasks.add(new Todo(keywords[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(false, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        } else if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("deadline")) {
+            try {
+                String[] details = keywords[1].split(" /by ");
+                isEmptyDescription = false;
+                tasks.add(new Deadline(details[0], details[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(isEmptyDescription, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        } else if (keywords[INDEX_OF_KEYWORD_OF_TASK].equalsIgnoreCase("event")) {
+            try {
+                String[] details = keywords[1].split(" /at ");
+                isEmptyDescription = false;
+                tasks.add(new Event(details[0], details[1]));
+                hasNoException = true;
+            } catch (ArrayIndexOutOfBoundsException e) {
+                printIncorrectInput(isEmptyDescription, keywords[INDEX_OF_KEYWORD_OF_TASK]);
+            }
+        }
+        if (hasNoException) {
+            printAddedTasked();
+            updateFile();
+        }
+    }
+
+    public void listTasks() {
+        printDividerLine();
+        System.out.println("\t Here are the tasks in your list:");
+        for (int i = 0; i < Task.getNumberOfTasks(); i++) {
+            System.out.println("      " + (i + 1) + "." + tasks.get(i));
+        }
+        printDividerLine();
+    }
+
+    public void markAsDone(String index) {
+        try {
+            int indexInteger = Integer.parseInt(index);
+            tasks.get(indexInteger - 1).setDone();
+            printDividerLine();
+            System.out.println("\tNice! I've marked this task as done:");
+            System.out.println("\t  " + tasks.get(indexInteger - 1));
+            printDividerLine();
+        } catch (NumberFormatException e) {
+            printDividerLine();
+            System.out.println("\t ☹ OOPS!!! The input after done must be a number.");
+            printDividerLine();
+        } catch (NullPointerException | IndexOutOfBoundsException e) {
+            printDividerLine();
+            System.out.println("\t ☹ OOPS!!! The task does not exist.");
+            printDividerLine();
+        }
+        updateFile();
+    }
+
+    public void deleteTask(String index) {
+        try {
+            int indexInteger = Integer.parseInt(index);
```

Comment:
Same suggestion. If you do -1 in this line, you don't have to do it everywhere else in this method.
`int indexInteger = Integer.parseInt(index) - 1;`